tools/common/Application.cs

```diff
@@ -495,21 +498,46 @@ public void RunRegistrar ()
 
 			var ps = new ReaderParameters ();
 			ps.AssemblyResolver = resolver;
-			resolvedAssemblies.Add (ps.AssemblyResolver.Resolve (AssemblyNameReference.Parse ("mscorlib"), new ReaderParameters ()));
-
-			var rootName = Path.GetFileNameWithoutExtension (RootAssembly);
-			if (rootName != Driver.GetProductAssembly (this))
-				throw ErrorHelper.CreateError (66, "Invalid build registrar assembly: {0}", RootAssembly);
+			resolvedAssemblies.Add ("mscorlib", ps.AssemblyResolver.Resolve (AssemblyNameReference.Parse ("mscorlib"), new ReaderParameters ()));
+
+			var productAssembly = Driver.GetProductAssembly (this);
+			bool foundProductAssembly = false;
+			foreach (var asm in RootAssemblies) {
+				var rootName = Path.GetFileNameWithoutExtension (asm);
+				if (rootName == productAssembly)
+					foundProductAssembly = true;
+				
+				try {
+					AssemblyDefinition lastAssembly = ps.AssemblyResolver.Resolve (AssemblyNameReference.Parse (rootName), new ReaderParameters ());
+					if (lastAssembly == null)
+						ErrorHelper.CreateWarning (7, "The root assembly '{0}' does not exist", rootName);
```

Comment:
You'll want to `continue` here, otherwise you'll run into a NullReferenceException further below.